platform-sdk/swirlds-common/src/main/java/com/swirlds/common/concurrent/internal/DefaultExecutorFactory.java

```diff
@@ -0,0 +1,156 @@
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Supplier;
+
+/**
+ * A default implementation of {@link ExecutorFactory}.
+ */
+public class DefaultExecutorFactory implements ExecutorFactory {
+
+    /**
+     * The thread factory for creating threads for executors.
+     */
+    private final ThreadFactory executorThreadFactory;
+
+    /**
+     * The thread factory for creating threads for scheduled executors.
+     */
+    private final ThreadFactory scheduledExecutorThreadFactory;
+
+    /**
+     * The thread factory for creating single threads.
+     */
+    private final ThreadFactory singleThreadFactory;
+
+    /**
+     * The thread factory for creating fork join worker threads.
+     */
+    private final ForkJoinWorkerThreadFactory forkJoinWorkerThreadFactory;
+
+    /**
+     * The uncaught exception handler for threads.
+     */
+    private final UncaughtExceptionHandler handler;
+
+    /**
+     * Create a new instance of {@link DefaultExecutorFactory}.
+     *
+     * @param singleThreadFactory            the thread factory for creating single threads
+     * @param executorThreadFactory          the thread factory for creating threads for executors
+     * @param scheduledExecutorThreadFactory the thread factory for creating threads for scheduled executors
+     * @param forkJoinWorkerThreadFactory    the thread factory for creating fork join worker threads
+     * @param handler                        the uncaught exception handler for threads
+     */
+    public DefaultExecutorFactory(
+            @NonNull final ThreadFactory singleThreadFactory,
+            @NonNull final ThreadFactory executorThreadFactory,
+            @NonNull final ThreadFactory scheduledExecutorThreadFactory,
+            @NonNull final ForkJoinWorkerThreadFactory forkJoinWorkerThreadFactory,
+            @NonNull final UncaughtExceptionHandler handler) {
+        this.singleThreadFactory = Objects.requireNonNull(singleThreadFactory, "singleThreadFactory must not be null");
+        this.executorThreadFactory =
+                Objects.requireNonNull(executorThreadFactory, "executorThreadFactory must not be null");
+        this.scheduledExecutorThreadFactory = Objects.requireNonNull(
+                scheduledExecutorThreadFactory, "scheduledExecutorThreadFactory must not be null");
+        this.forkJoinWorkerThreadFactory =
+                Objects.requireNonNull(forkJoinWorkerThreadFactory, "forkJoinWorkerThreadFactory must not be null");
+        this.handler = Objects.requireNonNull(handler, "handler must not be null");
+    }
+
+    @Override
+    public ForkJoinPool createForkJoinPool(int parallelism) {
+        if (parallelism <= 0) {
+            throw new IllegalArgumentException("parallelism must be greater than 0");
+        }
+        return new ForkJoinPool(parallelism, forkJoinWorkerThreadFactory, handler, false);
+    }
+
+    @Override
+    public ExecutorService createExecutorService(int threadCount) {
```

Comment:
nullity annotations for return types within this class

Reply:
The interface already has that definitions.